src/index.py

```diff
@@ -75,89 +75,55 @@ def _get_key(metadata: dict, key: str) -> str:
     return docs_value
 
 
-def _read_docs_index(base_path: Path) -> str:
+def _read_docs_index(base_path: Path) -> str | None:
```

Comment:
Wouldn't you rather use `Optional[str]`?

Reply:
Since Python 3.10 syntax for `Union` using the `|` operator has been introduced which has led to developers generally preferring `X | None` instead of `Optional[X]` because (1) it is more explicit in what it means, optional is a bit fuzzy as it could be interpreted as "this variable may not be defined" or "this variable could be `None`" and (2) doesn't require a `typing` import. What do you think? `Optional[X]` is definitely better than `Union[X, None]` for before Python 3.10

Reply:
Thx for the clarification